### DISCUSSION TITLE
way to style single dropdown

### DISCUSSION THREAD

**Discussion by vytautas-pranskunas-** [taiga-family/taiga-ui#9246] [Q&A] [answered]
i need to style just one dropdown, remove bg color, remove border, etc... how to do it not affecting default ones? Also change slide animation or disable it at all?
thanks

**Comment by waterplea** [comment, marked as answer]
To change CSS you can use `tuiDropdownAppearance="my-dropdown"` or provide it as DI options and then style it in global/unencapsulated styles:
```css
tui-dropdown[data-appearance="my-dropdown"] {
  ...
}
```
If you want to change animations you would need to provide your own entire component as `TUI_DROPDOWN_COMPONENT` since Angular Animations are not dynamically configurable. You can probably extend our component so you do not have to bother with positioning:
https://github.com/taiga-family/taiga-ui/blob/main/projects/core/directives/dropdown/dropdown.component.ts

**Comment by vytautas-pranskunas-** [comment]
i will do something. However what i quikli saw is - // @bad TODO: OnPush 
ufff :) our entire project is build with OnPush...

**Reply by waterplea** [reply]
Entire Taiga UI is built with OnPush too, that's why there's a TODO to make this component OnPush too :) But it's a bit tricky in its case because it has to be able to communicate change detection in 2 directions. Dropdowns show a template and whenever change detection is ran in the dropdown — it has to also run in place of template declaration and vice versa. There's only a handful of components that use Default strategy in Taiga UI

**Comment by vytautas-pranskunas-** [comment]
Observable as comunicator?

**Reply by waterplea** [reply]
Please reply to a comment, not create a new one, otherwise it's unclear how the conversation progresses. Don't forget that our components are general use flexible UI low level elements, they need to work in all scenarios. Another case where we use Default is dialogs, otherwise we will force all content that people might use for them to also be OnPush, and it should be up to the user, not enforced by us.

**Reply by vytautas-pranskunas-** [reply]
> Please reply to a comment, not create a new one, otherwise it's unclear how the conversation progresses. Don't forget that our components are general use flexible UI low level elements, they need to work in all scenarios. Another case where we use Default is dialogs, otherwise we will force all content that people might use for them to also be OnPush, and it should be up to the user, not enforced by us.

Ah isee. Sry i reply straight from gmail.
But it has little but implications to people that enforces onPush + as you know with signals and zonless in near future there will be only onPush
